### ISSUE TITLE
match_area / area_lookup get out of date

### ISSUE THREAD

**Issue by Moohan** [Public-Health-Scotland/phsmethods#71]
Probably one for @jackhannah95 as he wrote all the fancy SPARQL code!

The intention with area_lookup was that it should be updated with every package release but there haven't been any package releases. There's also the issue of people not updating to the latest version of `phsmethods`.

Could the solution be to have `match_area` do the API query live? Ideally, it could cache data so only one call would usually be needed. Another option (not sure how possible these are) could be to include data with the package then check how old it is (or even better check if there is a new version available at the API). If the data in the package is old; use the live call instead.

**Comment by jackhannah95** [comment]
Not me I'm afraid - @jvillacampa wrote the SPARQL.

At the time we decided not to have the API query built into the function because we were concerned about performance. The query can be quite slow to run at times on the server and there are the expected timeout issues with the VPN on desktop. 

It doesn't solve the longer-term issue of how to keep the area lookup up to date but maybe a one-off update of it would provide a temporary fix.

**Comment by Moohan** [comment]
Yeah, the function is a really nice idea but keeping the data behind it current is a really thorny issue!

We could probably set up a GitHub action to regularly (daily/weekly/monthly) run the scripts in data-raw then commit or open a PR to commit changes... I've used actions a bit before on some other repos / packages but never anything that 'custom'.

**Comment by jvillacampa** [comment]
Good point raised! I remember this was one of the things Jack and I discussed the most when creating the function. I think the idea of the GitHub action sounds very good (but never done it I am afraid). We could add a warning message to advice users to update their version of the package when using this function to ensure they have the latest lookup?

**Comment by Moohan** [comment]
I have no time right now but will hopefully have some over Christmas... would you want to look at a warning if I looked at a GH action (I'd first look at opening a monthly PR I think).
Also aware that I have at least 3 other things I've said I'd contribute to in this package, none of which are finished 😛
